### PR TITLE
fix: missing logs on ci timeout

### LINKED ISSUE #3424: No logs uploaded on timeout (CI)
See https://github.com/dragonflydb/dragonfly/actions/runs/10195266076

### PR DESCRIPTION
**Pull request by kostasrim**
Resolve #3424

The `env` variables exported when regression tests timeout are not working properly and the `if statement` on the action step `Print last log on timeout` would fail to read and upload the files set in `/tmp/last_log_file.txt`. Furthermore, another problem is the `job.timeout` argument that kills the whole job/matrix before the upload log step has a chance to run. For that, we need manual `timeouts` on the workflow similar to what we do in `regression tests action`. 

The fixes are simple:

* remove print last log on timeout action step
* copy the logs on timeout directly within the timeout step
* replace global timeout on CI workflow with timeout command per step

see also:
https://graphite.dev/guides/github-actions-timeouts#3-waiting-for-user-input-or-external-events

### CODE REVIEW

**Review by chakaz — APPROVED** [review]

File: tests/dragonfly/conftest.py [hunk 1]
```diff
@@ -378,6 +379,10 @@ def copy_failed_logs(log_dir, report):
             logging.error(f"🪵🪵🪵🪵🪵🪵 {file} 🪵🪵🪵🪵🪵🪵")
             shutil.copy(file, test_failed_path)
 
+    # Clean up
+    last_logs = open(LAST_LOGS, "w")
+    last_logs.close()
```
Comment (BorysTheDev) [inline tests/dragonfly/conftest.py:384]: why not remove it?

**Review by BorysTheDev — DISMISSED** [review]

File: tests/dragonfly/conftest.py (hunk 1 above)
Comment (kostasrim) [reply tests/dragonfly/conftest.py]: +1

**Comment by kostasrim** [comment]
@BorysTheDev @chakaz I fixed another issue as well :) 

We should not see any missing logs from now on.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -48,7 +48,6 @@ jobs:
             cxx_flags: ""
 
     runs-on: ubuntu-latest
-    timeout-minutes: 60
```
Comment (kostasrim) [inline .github/workflows/ci.yml:51]: root of all kinds of evil

Comment (BorysTheDev) [reply .github/workflows/ci.yml:51]: is it possible to stuck without timeout?

Comment (kostasrim) [reply .github/workflows/ci.yml:51]: no because we run each test with `timeout`

**Review by BorysTheDev — APPROVED** [review]

**Pull request merged**